internal/core/workers/operation_execution_worker/metrics.go

```diff
@@ -0,0 +1,105 @@
+package operation_execution_worker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/topfreegames/maestro/internal/core/monitoring"
+)
+
+const (
+	LabelNoOperationExecutorFound = "no_operation_executor_found"
+	LabelShouldNotExecute         = "should_not_execute"
+	LabelNextOperationFailed      = "next_operation_failed"
+	LabelStartOperationFailed     = "start_operation_failed"
+)
+
+var (
+	operationExecutionLatencyMetric           *prometheus.HistogramVec
+	operationOnErrorLatencyMetric             *prometheus.HistogramVec
+	operationEvictedCountMetric               *prometheus.CounterVec
+	operationExecutionWorkerFailedCountMetric *prometheus.CounterVec
+)
+
+func init() {
+	resetMetrics()
+}
+
+func resetMetrics() {
+	if operationExecutionLatencyMetric != nil {
+		prometheus.Unregister(operationExecutionLatencyMetric)
```

Comment:
Not sure why do we need to do this. Using `promauto` must be enough to guarantee a unique definition.

Reply:
The intention here is not to guarantee the unique definition. But, to only unregister metrics if there's at least one registered.

Reply:
Why do we need to do this? If we left them as package variables, they would be initialized only once.

Reply:
There's no way to clear the metrics between tests, the only way to do it is to unregister and register a new metric in place. That's why we're doing it.